src/main/java/user/delete_comment/DeleteCommentInteractor.java

```diff
@@ -65,7 +66,7 @@ public DeleteCommentResponseModel createResponseModel(DeleteCommentRequestModel
         // Handler will delete the specific comment from the ArrayList, will update the parent
         // with updated list of comments.
 
-        DeleteCommentHandler handler = new DeleteCommentHandler(parentType, commentId, comments);
+        DeleteCommentHandler handler = new DeleteCommentHandler(commentId, comments);
```

Comment:
Sorry I think I didn't make my self too clear. What I was trying to say is in the interactor, you can have `if (parentType == "review") {dataAccess.updateReview(...)} else if (parentType =="comment") {dataAccess.updateComment(...)}`. Similar for above when getting from db. You can have `if(parentType == "review"){dataAccess.getReview(...)} else if ...`. This means you only need four methods in your DAO: `getComment`, `getReview`, `updateComment`, `updateReview`. In the future, we may move all these checking to the controller and call the interactor with the correct DAO, but for this PR you can just do these checking in the interactor.

Reply:
OK, but a problem that I have right now is that because Comment and Review are not related by any inheritance, I have to make two separate variables to store them (in my if else statement, I need to two variables depending on if the parent is a Comment or a Review). The problem with this is that I need if else statements everywhere else in my interactor, because the getComments method may not work, because one of the two variables is going to be null (I need two variables in the if else statement, but only one of them is going to be an actual Comment or Review; the other variable will just be null) So in my code, the error I keep getting is that one of the variables may not have been initialized, and it makes checking quite messy. For now, I think the only way around this is to do the checking in the data access class, because there I can have separate helper functions for each case if the parent is a Comment or Review. So in essence, my interactor doesn't deal with if the parent is Comment or Review; all that is done in the data access (I know that might not be what you want, but because Comment and Review are of two different types, it is very hard to do the checking in the interactor). This way I think also works, I have the same 4 methods you outlined above, but I just have two extra, which are to check if the parent type is a Review or Comment. This way also just means that my interactor never actually has access to the parent; it just has access to the comments of the parent, which I think is also good because I shouldn't be able to access something that I do not need too. (My use case only needs to access to the comment being deleted and the list of comments)